tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -15,6 +15,15 @@ K_THREAD_STACK_DEFINE(mem_domain_2_stack, MEM_DOMAIN_STACK_SIZE);
 K_THREAD_STACK_DEFINE(mem_domain_6_stack, MEM_DOMAIN_STACK_SIZE);
 struct k_thread mem_domain_1_tid, mem_domain_2_tid, mem_domain_6_tid;
 
+#if defined(CONFIG_X86)
+K_MEM_PARTITION_DEFINE(wrong_part, 1, 0, K_MEM_PARTITION_P_RW_U_RW);
+#elif defined(CONFIG_ARM)
+K_MEM_PARTITION_DEFINE(wrong_part, 64, 32, K_MEM_PARTITION_P_RW_U_RW);
+#elif defined(CONFIG_ARC)
+K_MEM_PARTITION_DEFINE(wrong_part, 64, 32, K_MEM_PARTITION_P_RW_U_RW);
+#else
```

Comment:
This seems suspicious.  Why do different architectures have different needs here?  I'm confused about how this test operates.  You want to get an assertion failure, but don't test for that happening anywhere I can see?

If this is just testing that an __ASSERT condition trips, then that seems needless (the point to assertions is that they never happen!).   If there are code coverage tools that are flagging dead code due to assertions, then we should refactor to fix that instead of adding a test.  If you feel we need testing of this situation specifically, then it should be a real fatal runtime error and not an assert.

Reply:
Yes, I tried to write test to test assertion in code for code coverage. Seems that is not a way, and Andrew said about that in a comment below.